pyrit/orchestrator/crescendo_orchestrator.py

```diff
@@ -0,0 +1,239 @@
+from pyrit.orchestrator import Orchestrator
+from pyrit.prompt_target import CrescendoLLM
+from pyrit.score import CrescendoJudge
+import os
+import json
+# from langchain_openai import AzureChatOpenAI
+from langchain.schema import AIMessage
+from langchain.schema import HumanMessage
+# from transformers import AutoModelForCausalLM
+# from vllm import LLM, SamplingParams
+# from transformers import LlamaForCausalLM, AutoTokenizer
+# import time
+# from azure.ai.contentsafety.models import TextCategory
+# from azure.core.exceptions import HttpResponseError
+# from azure.ai.contentsafety.models import AnalyzeTextOptions
+
+
+class CrescendoOrchestrator(Orchestrator):
+
+    def __init__(
+        self,
+        target_task: str,
+        target_llm: CrescendoLLM,
+        attack_llm: CrescendoLLM,
+        judge_llm:  CrescendoJudge,
+        metaPromptsPath: str,
+        crescendo_variant: int = 1,
+        max_rounds: int = 10,
+        verbose: bool = False,
+    ) -> None:
+        super().__init__(
+            verbose=verbose
+        )
+        self._target_task = target_task
+        self._max_rounds = max_rounds
+        self._target_llm = target_llm
+        self._attack_llm = attack_llm
+        self._judge_llm = judge_llm
+        self._metaPromptFile = metaPromptsPath+'crescendomationVariant%i.txt'%crescendo_variant
+
+        # Prompt templates
+
+
+
+    # The Crescendo automation tool
+    def autoCrescendo(self,folderName, fileName,verbose=False,useACF=False,usePerspectiveAPI=True):
+        # Variable holding the previous questions, their summaries, results in terms of the Judge score and flag.
+        questionsWithSummary = []
+        # Variable holding everything happening including questions, their responses, the different evaluation scores, and the reasoning behind the jailbreak
+        allInOne = []
+        lastResponse = ''
+        pathToSave = './%s'%folderName
+        # The history of the conversation for the target model
+        history = [] 
+        resultFlag = False
+
+        if os.path.isdir(pathToSave):  
+            print(f'A folder for {pathToSave} already exists.' )
+        else:
+            os.makedirs(pathToSave)
+
+        summaryOfQuestion = ''
+        counter = 0
+        counter2 = 0
+        i = 0
+
+        # The main loop of executing Crescendo for 'numOfRounds' rounds
+        while (i<self._max_rounds):
+            i = i+1
```

Comment:
```suggestion
        # The main loop of executing Crescendo for 'numOfRounds' rounds
        for i in range(1, self._max_rounds + 1):
```
A more descriptive name than `i` is also best. Maybe iteration?